crates/ntx-builder/src/builder/server/state.rs

```diff
@@ -81,8 +81,13 @@ impl PendingNotes {
     /// Move the notes whose nullifiers belong to the input `nullifiers` into the in-flight
     /// set for the given `tx_id`, removing them from the indices and pruning empty tag buckets and
     /// queue entries.
-    pub fn insert_inflight(&mut self, tx_id: TransactionId, nullifiers: &[Nullifier]) {
-        if nullifiers.is_empty() {
+    pub fn insert_inflight(
+        &mut self,
+        tx_id: TransactionId,
+        nullifiers: impl IntoIterator<Item = Nullifier>,
```

Comment:
Not sure why we are using an iterator here instead of the slice.

Reply:
I think this is more flexible, but . Specifically, RPC endpoints and executed transactions both yield owned collections that can be mapped to a nullifier iterator.